pythonFiles/include/blender_vscode/load_addons.py

```diff
@@ -11,25 +12,30 @@
 from .utils import is_addon_legacy
 
 
-def setup_addon_links(addons_to_load: list[AddonInfo]):
-
-    path_mappings = []
+def setup_addon_links(addons_to_load: List[AddonInfo]) -> List[Dict]:
+    path_mappings: List[Dict] = []
 
     for addon_info in addons_to_load:
-        user_addon_directory = get_user_addon_directory(Path(addon_info.load_dir))
-        print(f"USER ADDON: {user_addon_directory}")
-
-        if not os.path.exists(user_addon_directory):
-            os.makedirs(user_addon_directory)
-
-        if is_addon_legacy(Path(addon_info.load_dir)) and not str(user_addon_directory) in sys.path:
-            sys.path.append(str(user_addon_directory))
-
-        if is_in_any_addon_directory(addon_info.load_dir):
-            load_path = addon_info.load_dir
+        default_directory = get_user_addon_directory(Path(addon_info.load_dir))
+        if is_addon_legacy(Path(addon_info.load_dir)):
+            if is_in_any_addon_directory(addon_info.load_dir):
+                # blender knows about addon and can load it
+                load_path = addon_info.load_dir
+            else:  # is in external dir of is in extensions dir
+                load_path = os.path.join(default_directory, addon_info.module_name)
+                if str(load_path) not in sys.path:
+                    sys.path.append(str(load_path))
```

Comment:
I am not sure when adding path is required

Reply:
adding sys.path for extensions resylts in blneder warning about policy violation